application-tour-test/application-tour-test-tests/src/test/it/org/xwiki/contrib/tour/test/ui/TourApplicationTest.java

```diff
@@ -196,39 +196,41 @@ private void tourBoundToPage() throws Exception
         assertFalse(secondPage.isTourDisplayed());
         assertTrue(secondPage.hasResumeButton());
 
-        // Resume (to step 4)
+        // Resume (to step 1, as we have ended the tour)
```

Comment:
I'm assuming that you've changed the behavior slightly:
* If the tour is ended, it goes back to step 1 when restarted
* If the tour is closed, it resumes back to where it was left when restarted

Sounds ok to me.